src/nmea.cpp

```diff
@@ -0,0 +1,1043 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2020 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file nmea.cpp
+ *
+ * NMEA protocol implementation.
+ *
+ * @author WeiPeng Guo <guoweipeng1990@sina.com>
+ * @author Stone White <stone@thone.io>
+ * @author Jose Jimenez-Berni <berni@ias.csic.es>
+ *
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <math.h>
+#include <string.h>
+#include <ctime>
+
+#include "nmea.h"
+
+#ifndef M_PI_F
+# define M_PI_F 3.14159265358979323846f
+#endif
+
+#define MAX(X,Y)    ((X) > (Y) ? (X) : (Y))
+#define NMEA_UNUSED(x) (void)x;
+
+GPSDriverNMEA::GPSDriverNMEA(GPSCallbackPtr callback, void *callback_user,
+			     sensor_gps_s *gps_position,
+			     satellite_info_s *satellite_info,
+			     float heading_offset) :
+	GPSHelper(callback, callback_user),
+	_gps_position(gps_position),
+	_satellite_info(satellite_info),
+	_heading_offset(heading_offset)
+{
+	decodeInit();
+}
+
+/*
+ * All NMEA descriptions are taken from
```

Comment:
Is this accurate?

Reply:
This comes from the original source code developed by DimianZhan:add_nmea_driver I am not sure how to do the right author attribution (as I couldn't branch from that one).

Reply:
I would just remove it then, as it's likely a copy-paste leftover and possibly add your reference